### DISCUSSION TITLE
Use `strict: true` for shared `tsconfig/next.json`

### DISCUSSION THREAD

**Discussion by timc1** [vercel/turborepo#816] [Help] [answered]
Really enjoy using Turborepo :-)

I used the `create-turbo` command to bootstrap an app and was running into some missing type errors while developing within `/web`. Turned out the tsconfig for our Next.js apps under `/apps/*` have strict type checking [off by default](https://github.com/vercel/turborepo/blob/b95a07e147c93b50f4927335d4914e5e5e82a97d/packages/create-turbo/templates/_shared_ts/packages/tsconfig/nextjs.json#L10), where flipping the config to `true` solved my issues.

I'm not too familiar with this so wanted to make sure if there was an explicit reason for this config that I may be missing. Thanks again!

**Comment by PaulRBerg** [comment]
I just bumped into another case like this - I spent a few good hours debugging, ultimately realizing that the culprit was `strict` set to `false`.

https://github.com/vercel/turbo/blob/2baad6fa7a3b3fda35130f9c53cedcb724fc931c/examples/basic/packages/tsconfig/nextjs.json#L16

What is the rationale behind this decision? The overwhelming majority of Next.js templates out there have `strict` enabled, including the official template created via `create-next-app`.

**edit**: I think I have figured it out - the rationale is that Next.js doesn't use strict mode by default, as per the docs on [`reactStrictMode`](https://nextjs.org/docs/pages/api-reference/next-config-js/reactStrictMode). Right?

**Comment by anthonyshew** [comment, marked as answer]
No explicit reason. This was most likely an oversight when the examples were originally written.

I'm fixing this in #6762. Thank you for calling it out!
